trunner/runners/Stm32l4Runner.py

```diff
@@ -0,0 +1,77 @@
+#
+# Phoenix-RTOS test runner
+#
+# Stm32l4 runner
+#
+# Copyright 2021 Phoenix SYstems
+# Authors: Mateusz Niewiadomski
+#
+import subprocess
+import serial
+import time
+import pexpect
+import pexpect.fdpexpect
+
+from .common import DeviceRunner
+from .common import _BOOT_DIR
+
+
+class Stm32l4Runner(DeviceRunner):
+
+    class oneByOne_fdspawn(pexpect.fdpexpect.fdspawn):
+
+        def send(self, s):
+            for c in s:
+                super().send(c)
+                time.sleep(0.03)
+
+        def sendline(self, s):
+            self.send(s)
+            super().send('\n')
+
+    OPENOCD_CMD = ['openocd -f interface/stlink.cfg -f target/stm32l4x.cfg -c \
+"reset_config srst_only srst_nogate connect_assert_srst" -c "program ', ' 0x08000000 verify reset exit"']
+    IMAGE = 'phoenix-armv7m4-stm32l4x6.bin'
+
+    def __init__(self, port1, port2=None):
+        super().__init__(port1)
+        self.port2 = port2
+        self.image_flashed = False
+        self.openocd_cmd = self.OPENOCD_CMD[0] + str(_BOOT_DIR / self.IMAGE) + self.OPENOCD_CMD[1]
```

Comment:
nah, the filepath is not correctly escaped (if there is a space in the path it would fail).

I would go for providing arguments as an array to `subprocess.Popen()`

Reply:
used:
```
BINARY_PATH = path.join(_BOOT_DIR ,'phoenix-armv7m4-stm32l4x6.bin')
```

passing arguments to `Popen()` as array seems to be problematic as there are `"` and for some reason escaping them or not makes no difference even using `shlex`